server/src/main/java/io/crate/metadata/view/ViewInfo.java

```diff
@@ -45,14 +45,21 @@ public class ViewInfo implements RelationInfo {
     private final RelationName ident;
     private final String definition;
     private final List<Reference> columns;
+    private final List<Reference> references;
     private final String owner;
     private final SearchPath searchPath;
 
     @VisibleForTesting
-    public ViewInfo(RelationName ident, String definition, List<Reference> columns, @Nullable String owner, SearchPath searchPath) {
+    public ViewInfo(RelationName ident, String definition, List<Reference> references, @Nullable String owner, SearchPath searchPath) {
         this.ident = ident;
         this.definition = definition;
-        this.columns = columns;
+        this.references = references
+            .stream()
+            .sorted(Reference.CMP_BY_POSITION_THEN_NAME)
+            .toList();
+        this.columns = this.references.stream()
+            .filter(r -> r.column().isRoot())
```

Comment:
afaik this should be sorted as well
```suggestion
            .filter(r -> r.column().isRoot())
            .sorted(Reference.CMP_BY_POSITION_THEN_NAME)
```

Reply:
Merging as is to avoid extra sorting - in case I missed smth will follow up on it.

Reply:
ah ok missed this.